pkg/build/registry/buildconfig/etcd/etcd.go

```diff
@@ -47,3 +51,32 @@ func NewStorage(s storage.Interface) *REST {
 
 	return &REST{store}
 }
+
+func (r *REST) Delete(ctx kapi.Context, name string, options *kapi.DeleteOptions) (runtime.Object, error) {
+	bcObj, err := r.Get(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+	bc := bcObj.(*api.BuildConfig)
+
+	// Just update DeletionTimestamp so Builds get deleted
+	if bc.DeletionTimestamp.IsZero() {
+		now := unversioned.Now()
+		bc.DeletionTimestamp = &now
+		bcObj, _, err = r.Update(ctx, bc)
+		return bcObj, err
+	}
+
+	if !bc.Status.CanDelete {
+		err = kapierrors.NewConflict("BuildConfig", fmt.Sprintf("%s/%s", bc.Namespace, bc.Name), fmt.Errorf("The system is ensuring all Builds instantiated from this BuildConfig are removed. Upon completion, this BuildConfig will automatically be purged by the system."))
+		return nil, err
```

Comment:
What do you mean by no-oping it? I think it is kind of an error, because the BC did not get deleted. And I'm not sure, but I suspect it'll get retried thanks to us returning error. Also, the namespace code does the same.